server/repository/migrations/postgres/2021-11-05T10-00_create_sync_buffer_table/up.sql

```diff
@@ -0,0 +1,15 @@
+CREATE TYPE sync_action AS ENUM (
+    'UPSERT',
+    'DELETE',
+    'MERGE'
+);
+
+CREATE TABLE sync_buffer (
+    record_id TEXT NOT NULL PRIMARY KEY,
+    received_datetime TIMESTAMP NOT NULL,
+    integration_datetime TIMESTAMP,
```

Comment:
integration_datetime should be set when integration_error happens (we don't have OR filter yet)

Reply:
665ee6d82f630d1be2a7505ef5d5bbd8c973b7bf in https://github.com/openmsupply/open-msupply/issues/338